tests/rptest/scale_tests/extreme_recovery_test.py

```diff
@@ -0,0 +1,237 @@
+        """Validate all data after recovery."""
+        self.logger.info(
+            f"vallidate_cluster base:restored: {baseline}:{restored}")
+
+        self._validate_partition_last_offset()
+        expected_topics = [
+            topic.name for topic in self.expected_recovered_topics
+        ]
+
+        # consume and validate all data
+        t0 = time()
+        consumers: list[FranzGoVerifiableSeqConsumer] = list()
+        for topic in self.topics:
+            self.logger.info(
+                f'validate_cluster: starting consumer for "{topic}"')
+            c = FranzGoVerifiableSeqConsumer(self.ctx, self.redpanda, topic,
+                                             self.MSG_SIZE)
+            c.start()
+            consumers.append(c)
+
+        total_bytes = 0
+        for c in consumers:
+            self.logger.info(
+                f'validate_cluster: waiting on consumer for "{c._topic}"')
+            c.shutdown()
+            c.wait(self.expected_transfer_sec)
+            total_bytes += c.consumer_status.valid_reads * self.MSG_SIZE
+            assert c.consumer_status.invalid_reads == 0, "No invalid reads expected."
+            assert c.consumer_status.valid_reads == self._msg_per_topic
+
+        delta_t = time() - t0
+        self._status(
+            f'validate_cluster: consumed {total_bytes/2**20:0.1f} MB in {delta_t:0.3f} sec;'
+            f' {(total_bytes/delta_t)/2**20:0.3f} MB/s.')
+
+    @property
+    def second_restart_needed(self):
+        """Return True if the restart is needed afer first validation steps"""
+        # XXX TODO
+        return False
+
+    def after_restart_validation(self):
+        """Check that topic is writable after restart"""
+        # XXX TODO
+
+
+# "It's like, totally extreme, bro."
+# Try 1: hack topic recovery test into being a scale test.
+class ExtremeRecoveryTest(TopicRecoveryTest):
+    def __init__(self, test_context: TestContext):
+        extra_rp_conf = {"blah": "bloo"}
+        super(ExtremeRecoveryTest,
+              self).__init__(test_context,
+                             num_brokers=RecoveryScale.NODE_COUNT)
+        assert (self.si_settings and
+                self.si_settings.cloud_storage_max_connections > 10), \
+                    "Need more connections for scale test"
+        self.initial_cleanup_needed = True
+        self.ctx = test_context
+
+    def tearDown(self):
+        super(ExtremeRecoveryTest, self).tearDown()
+
+    @cluster(num_nodes=8, log_allow_list=TRANSIENT_ERRORS)
+    def test_recovery_scale(self):
+        # This test requires dedicated system resources
+        assert self.redpanda.dedicated_nodes
+
+        assert self.s3_client
+        topics = [
+            TopicSpec(partition_count=RecoveryScale.part_per_topic,
+                      replication_factor=3)
+            for i in range(RecoveryScale.NUM_TOPICS)
+        ]
+
+        extremeRecovery = RecoveryScale(self.test_context, self.s3_client,
```

Comment:
nit: should be snake_case for consistency, feel free to add to a later PR

Reply:
Done in latest update, resolving.